test/helpers/foundry/foundryup.mts

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env -S node --require "./node_modules/tsx/dist/preflight.cjs" --import "./node_modules/tsx/dist/loader.mjs"
+
+import { join, relative } from 'node:path';
+import { homedir } from 'node:os';
+import { Dir, readFileSync } from 'node:fs';
+import { opendir, symlink, unlink, copyFile, rm } from 'node:fs/promises';
+import { createHash } from 'node:crypto';
+import { exit, cwd } from 'node:process';
+import { parse as parseYaml } from 'yaml';
+import {
+  BinFormat,
+  Platform,
+  extractFrom,
+  getVersion,
+  printBanner,
+  say,
+  parseArgs,
+  isCodedError,
+  noop,
+} from './helpers.mts';
+
+const parsedArgs = parseArgs();
+
+const { enableGlobalCache } = parseYaml(readFileSync('.yarnrc.yml', 'utf-8'));
+const CACHE_DIR = enableGlobalCache
+  ? join(homedir(), '.cache', 'metamask')
+  : join(cwd(), '.metamask', 'cache');
```

Comment:
Probably not necessary, but this ensures that we keep our own cache alongside the `.yarn` cache. Why? I had to put it somewhere, and copying whatever we do for yarn just felt right, I guess.